libshaderc_util/include/libshaderc_util/compiler.h

```diff
@@ -168,17 +167,23 @@ class Compiler {
   // Any errors are written to the error_stream parameter.
   // total_warnings and total_errors are incremented once for every
   // warning or error encountered respectively.
-  // Returns true if the compilation succeeded and the result could be written
-  // to output, false otherwise.
-  bool Compile(const shaderc_util::string_piece& input_source_string,
-               EShLanguage forced_shader_stage, const std::string& error_tag,
-               const std::function<EShLanguage(std::ostream* error_stream,
-                                               const shaderc_util::string_piece&
-                                                   error_tag)>& stage_callback,
-               const CountingIncluder& includer, std::ostream* output_stream,
-               std::ostream* error_stream, size_t* total_warnings,
-               size_t* total_errors,
-               GlslInitializer* initializer) const;
+  //
+  // Returns a tuple consists of three fields. 1) a boolean which is true when
+  // the compilation succeeded, and false otherwise; 2) a vector of 32-bit words
+  // which contains the compilation output data, either compiled SPIR-V binary
+  // code, or the text string generated in preprocessing-only or disassembly
+  // mode; 3) the size of the output data. When the output is SPIR-V binary
+  // code, the size will always be N times of four. If the output is a text
```

Comment:
For (3) say whether the size is in bytes or not.  And when it's a string, say whether the size includes a null terminator, or not.